### PR TITLE
Improved error reporting for invalid properties

### PR DESCRIPTION
**Pull request by petrochuk**
Expected error message:
Property 'NoValidControl' not found on type 'Microsoft.PowerPlatform.PowerApps.Persistence.Models.Control'.

### CODE REVIEW

File: src/Persistence.Tests/_TestData/InvalidYaml/screen-with-not-valid-control.pa.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Screens:
+- Screen: 
+  Name: Screen1
+  Properties:
+    LoadingSpinnerColor: =RGBA(56, 96, 178, 1)
+  Children:
+  - NoValidControl:
```
Comment (petrochuk) [inline src/Persistence.Tests/_TestData/InvalidYaml/screen-with-not-valid-control.pa.yaml:7]: @lesaltzm FYI

**Comment by mizrael** [comment]
I don't follow: shouldn't we be throwing when there's no template found?

**Comment by petrochuk** [comment]
> I don't follow: shouldn't we be throwing when there's no template found?

We don't know if this is template or unexpected property. We will be a little more strict for controls

**Comment by mizrael** [comment]
> > I don't follow: shouldn't we be throwing when there's no template found?
> 
> We don't know if this is template or unexpected property. We will be a little more strict for controls

why? if it's under the `Children` node it's definitely a Control. Maybe I am missing something ?

**Comment by petrochuk** [comment]
> > > I don't follow: shouldn't we be throwing when there's no template found?
> > 
> > 
> > We don't know if this is template or unexpected property. We will be a little more strict for controls
> 
> why? if it's under the `Children` node it's definitely a Control. Maybe I am missing something ?

Here is a little better example. What should we do in this case?

``` yaml
Screens:
- Screen: 
  Name: Screen1
  Properties:
    LoadingSpinnerColor: =RGBA(56, 96, 178, 1)
  Children:
  - NoValidControl:
    StillNotValid:
    AgianNotValid:
    NotValid:
    Name: Foo
    MyName: Bar
    Properties:
      BorderColor: =RGBA(0, 18, 107, 1)
      BorderThickness: =2

```

**Comment by mizrael** [comment]
took this offline. So this change is actually throwing when nodes are not recognized as valid templates.

Thanks!

**Review by mizrael — APPROVED** [review]

**Pull request merged**